Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/impl/apmlog/LogFilterControllerWithAPMLog.java

```diff
@@ -151,14 +152,21 @@ private Map<String, Object> getEditorOptionPayload(FilterType filterType,
         switch (filterType) {
             case CASE_SECTION_ATTRIBUTE_COMBINATION:
             case EVENT_ATTRIBUTE_DURATION:
+            case NODE_COST:
                 data = (String) param.get("data");
                 if (filterType == FilterType.EVENT_ATTRIBUTE_DURATION &&
                         !isValidAttributeDurationPayload(mainAttribute, data)) {
                     Messagebox.show(
                             parent.getLabel("failedFilterNodeDurationSingleValue_message"),
-                            parent.getLabel("common_error_text"), Messagebox.OK, Messagebox.ERROR);
+                            parent.getLabel(ERROR_TITLE_KEY), Messagebox.OK, Messagebox.ERROR);
                     return null;
                 }
```

Comment:
This condition is not used (always false) because the filterType is NODE_COST in this switch case.

Reply:
ah I see. This is for three switch cases, not one.

Reply:
This code is reached if filterType is either CASE_SECTION_ATTRIBUTE_COMBINATION,  EVENT_ATTRIBUTE_DURATION or NODE_COST so there is still a need to check.

Reply:
Can we move the condition check to its own switch case and no need to check filterType again?

Reply:
Can we move the error message handling outside and before the switch case? It is a separate part from the switch case handling. It's not easy to read switch...case and then if...then...else again within switch case (a bit circular).